src/app/api/auth/[...nextauth]/route.jsx

```diff
@@ -0,0 +1,45 @@
+import NextAuth from "next-auth/next";
+import googleProvider from "next-auth/providers/google";
+import { FirestoreAdapter } from "@auth/firebase-adapter";
+import { cert } from "firebase-admin/app";
+
+export const authOptions = {
+  // eslint-disable-next-line new-cap
+  adapter: FirestoreAdapter({
+    credential: cert({
+      projectId: process.env.NEXT_PUBLIC_PROJECT_ID,
+      clientEmail: process.env.NEXT_PUBLIC_CLIENT_EMAIL,
+      privateKey: process.env.NEXT_PUBLIC_PRIVATE_KEY,
+    }),
+  }),
+  providers: [
+    googleProvider({
+      profile(profile) {
+        return {
+          id: profile.sub,
+          role: "hacker",
```

Comment:
remove the hacker role, we can leave it at the default profile since no role means they just logged in and haven't registered, only once do they register do they get accepted do they become a hacker